data/transactions/logic/simulate_test.go

```diff
@@ -0,0 +1,731 @@
+		txnArray[i] = txn.Txn
+	}
+
+	client := makeTestClient()
+	groupID, err := client.GroupID(txnArray)
+	if err != nil {
+		return err
+	}
+
+	for i := range txgroup {
+		txgroup[i].Txn.Header.Group = groupID
+	}
+
+	return nil
+}
+
+func uint64ToBytes(num uint64) []byte {
+	ibytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(ibytes, num)
+	return ibytes
+}
+
+// ==============================
+// > Simulation Tests
+// ==============================
+
+func TestPayTxn(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	l := makeSimulationTestLedger()
+	s := v2.MakeSimulator(l)
+
+	accounts := makeTestAccounts()
+	sender := accounts[0].Address
+
+	txgroup := []transactions.SignedTxn{
+		{
+			Txn: transactions.Transaction{
+				Type:   protocol.PaymentTx,
+				Header: makeBasicTxnHeader(sender),
+				PaymentTxnFields: transactions.PaymentTxnFields{
+					Receiver: sender,
+					Amount:   basics.MicroAlgos{Raw: 0},
+				},
+			},
+		},
+	}
+
+	result, err := s.SimulateSignedTxGroup(txgroup)
+	require.NoError(t, err)
+	require.Empty(t, result.FailureMessage)
+}
+
+func TestOverspendPayTxn(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	l := makeSimulationTestLedger()
+	s := v2.MakeSimulator(l)
+
+	accounts := makeTestAccounts()
+	sender := accounts[0].Address
+	balances := makeTestBalances()
+
+	txgroup := []transactions.SignedTxn{
+		{
+			Txn: transactions.Transaction{
+				Type:   protocol.PaymentTx,
+				Header: makeBasicTxnHeader(sender),
+				PaymentTxnFields: transactions.PaymentTxnFields{
+					Receiver: sender,
+					Amount:   basics.MicroAlgos{Raw: balances[sender] + 100}, // overspend
+				},
+			},
+		},
+	}
+
+	result, err := s.SimulateSignedTxGroup(txgroup)
+	require.NoError(t, err)
+	require.Contains(t, *result.FailureMessage, "tried to spend {1000000100}")
+}
+
+func TestSimpleGroupTxn(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	l := makeSimulationTestLedger()
+	s := v2.MakeSimulator(l)
+
+	accounts := makeTestAccounts()
+	sender1 := accounts[0].Address
+	sender2 := accounts[1].Address
+
+	// Send money back and forth
+	txgroup := []transactions.SignedTxn{
+		{
+			Txn: transactions.Transaction{
+				Type:   protocol.PaymentTx,
+				Header: makeBasicTxnHeader(sender1),
+				PaymentTxnFields: transactions.PaymentTxnFields{
+					Receiver: sender2,
+					Amount:   basics.MicroAlgos{Raw: 1000000},
+				},
+			},
+		},
+		{
+			Txn: transactions.Transaction{
+				Type:   protocol.PaymentTx,
+				Header: makeBasicTxnHeader(sender2),
+				PaymentTxnFields: transactions.PaymentTxnFields{
+					Receiver: sender1,
+					Amount:   basics.MicroAlgos{Raw: 0},
+				},
+			},
+		},
+	}
+
+	// Should fail if there is no group parameter
+	result, err := s.SimulateSignedTxGroup(txgroup)
+	require.NoError(t, err)
+	require.Contains(t, *result.FailureMessage, "had zero Group but was submitted in a group of 2")
+
+	// Add group parameter
+	attachGroupID(txgroup)
```

Comment:
@jdtzmn Minor:  Is there a reason the returned error is ignored here rather than asserted for _no error_?

Reply:
There is not - thank you for spotting it.